yarn-project/end-to-end/src/e2e_inclusion_proofs_contract.test.ts

```diff
@@ -75,16 +77,44 @@ describe('e2e_inclusion_proofs_contract', () => {
     }
   });
 
+  it('note existence failure case', async () => {
+    // Owner of a note
+    const owner = AztecAddress.random();
+
+    const blockNumber = await pxe.getBlockNumber();
+    const randomNoteCommitment = Fr.random();
+    await expect(
+      contract.methods.proveNoteInclusion(owner, blockNumber, randomNoteCommitment).send().wait(),
+    ).rejects.toThrow(/Leaf value: 0x[0-9a-fA-F]+ not found in tree/);
```

Comment:
UX thing - why are the error messages so different from each other? If it isn't tooo much work, Can all of them be consistent:

For note failure - `Can't prove note inclusion in note hash tree for leaf value: 0xabc. Are you sure note exists?`
For public data - `Can't prove value exists in public data tree for value: 0xabc. Are you sure it exists?`
For nullifier inclusion failure - `Can't prove nullifer inclusion in nullifier tree for leaf value: 0xabc. Are you sure nullifier exists?` 

Actually Jan for nullifier tree inclusion, do we check against the low nullifier witness? 

For nullifier **non**inclusion failure -  `nullifier exists in the nullifier tree. Low nullifier [value].next_value is [nullifier]

Reply:
Made the errors clearer in https://github.com/AztecProtocol/aztec-packages/pull/3255/commits/edf4628c48a81c94e2e9f6c2d1f8065cf37ce496

For the note inclusion failure case it's having the same error is a bit tricky because there it fails in the oracle call and not when doing assertions and that's because in sparse tree and indexed tree you never fail to get a membership witness.

> Actually Jan for nullifier tree inclusion, do we check against the low nullifier witness?

No, for nullifier inclusion I just do standard membership proof.

Reply:
Is the standard to do it the low nullifier way in the rest of the codebase? I have not worked with indexed tree much before